app/src/main/java/science/apolline/view/activity/SplashScreen.kt

```diff
@@ -436,30 +435,36 @@ class SplashScreen : RootActivity(), AnkoLogger {
             val newMenu = CircleImageView(this)
             val currentDrawable = ContextCompat.getDrawable(this, drawableId)
             val willBeWhite = currentDrawable!!.constantState.newDrawable()
+            willBeWhite.mutate().setColorFilter(Color.WHITE, PorterDuff.Mode.SRC_ATOP)
 
-            if (bounded) {
-                newMenu.setBackgroundResource(R.drawable.circle_menu_shape_item_bounded)
-            } else {
-                newMenu.setBackgroundResource(R.drawable.circle_menu_shape_item)
+            // newMenu.setBackgroundResource(R.drawable.circle_menu_shape_item)
+            info("device to add : $name")
+            if (name.toLowerCase().contains(regex = "^ioio.".toRegex())) {
+                newMenu.setBackgroundResource(R.drawable.circle_menu_shape_item_ioio)
+                willBeWhite.mutate().setColorFilter(ContextCompat.getColor(this, R.color.primary_text), PorterDuff.Mode.SRC_ATOP)
+            } else if (name.toLowerCase().contains("^appa.".toRegex()))
```

Comment:
Regex and associated behaviours should be externatlized in an interface to ease the integration of future sensors

Reply:
At least extract as a constant in a first place.

Reply:
I need more specification please, what should i extract ? what i'm being asked is not very clear.

Reply:
Extract the magic strings `"^appa."` and `"^ioio."` as a constants instead of regex hidden in the code.

Reply:
oh okay i see